aospy/test/test_calc_basic.py

```diff
@@ -143,5 +146,55 @@ def setUp(self):
             'dtype_out_vert': 'vert_int'
         }
 
+class TestCalcAttrs(unittest.TestCase):
+    def test_units_description_attrs(self):
+        _test_attrs('', '', None, '', '')
+        _test_attrs('m', '', None, 'm', '')
+        _test_attrs('', 'rain', None, '', 'rain')
+        _test_attrs('m', 'rain', None, 'm', 'rain')
+        _test_attrs('', '', 'vert_av', '', '')
+        _test_attrs('m', '', 'vert_av', 'm', '')
+        _test_attrs('', 'rain', 'vert_av', '', 'rain')
+        _test_attrs('m', 'rain', 'vert_av', 'm', 'rain')
+        _test_attrs('', '', 'vert_int',
+                    '(vertical integral of quantity with unspecified units)',
+                    '')
+        _test_attrs('m', '', 'vert_int',
+                    '(vertical integral of m): m kg m^-2)',
+                    '')
+        _test_attrs('', 'rain', 'vert_int',
+                    '(vertical integral of quantity with unspecified units)',
+                    'rain')
+        _test_attrs('m', 'rain', 'vert_int',
+                    '(vertical integral of m): m kg m^-2)',
+                    'rain')
+
+def _test_attrs(units, description, dtype_out_vert, expected_units,
+                expected_description):
+    da = xr.DataArray(None)
+    ds = xr.Dataset({'bar': 'foo', 'boo': 'baz'})
+    _add_units_and_description_attrs(da, units, description, dtype_out_vert)
+    _add_units_and_description_attrs(ds, units, description, dtype_out_vert)
+    assert expected_units == da.attrs['units']
+    assert expected_description == da.attrs['description']
+    for name, d_arr in ds.data_vars.items():
+        assert expected_units == d_arr.attrs['units']
+        assert expected_description == d_arr.attrs['description']
+
+def _test_output_attrs(calc, file):
```

Comment:
Similar to above comment, I would move this to above the `TestCalcBasic` definition, since it's used within that class.